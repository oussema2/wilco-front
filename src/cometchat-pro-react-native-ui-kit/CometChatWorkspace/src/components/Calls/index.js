export { default as CometChatIncomingCall } from './CometChatIncomingCall';
export { default as CometChatOutgoingCall } from './CometChatOutgoingCall';
export { default as CometChatOutgoingDirectCall } from './CometChatOutgoingDirectCall';
export { default as CometChatIncomingDirectCall } from './CometChatIncomingDirectCall';
