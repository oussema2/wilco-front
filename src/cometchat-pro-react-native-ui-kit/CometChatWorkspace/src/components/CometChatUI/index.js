/* eslint-disable import/no-duplicates */
import React, { useEffect, useRef, useState } from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { View, Text } from 'react-native';
import { CometChatGroupListWithMessages } from '../Groups';
import { CometChatUserListWithMessages } from '../Users';
import { CometChatConversationListWithMessages } from '../Chats';
import { CometChatContextProvider } from '../../utils/CometChatContext';
import { CometChatUserProfile } from '../UserProfile';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import MCIIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import theme from '../../resources/theme';
import { heightRatio } from '../../utils/consts';


const Tab = createBottomTabNavigator();

function CometChatUI() {
  const [tabs, setTabs] = useState(null);
  const contextRef = useRef(null);

  useEffect(() => {
    checkRestrictions();
  }, []);
  const checkRestrictions = async () => {
    let isChatEnabled = await contextRef.current.state.FeatureRestriction.isRecentChatListEnabled();
    let isGroupListEnabled = await contextRef.current.state.FeatureRestriction.isGroupListEnabled();
    let isUserSettingsEnabled = await contextRef.current.state.FeatureRestriction.isUserSettingsEnabled();
    let isUserListEnabled = await contextRef.current.state.FeatureRestriction.isUserListEnabled();
    let isCallListEnabled = await contextRef.current.state.FeatureRestriction.isCallListEnabled();
    setTabs({
      isChatEnabled,
      isGroupListEnabled,
      isUserSettingsEnabled,
      isUserListEnabled,
      isCallListEnabled,
    });
  };
  return (
    <CometChatContextProvider ref={contextRef}>
      {tabs ? (
        <Tab.Navigator
          screenOptions={({ route }) => ({
            tabBarIcon: ({ color }) => {
              let iconName;

              if (route.name === 'Chats') {
                return (
                  <MCIIcons name="chat" size={25 * heightRatio} color={color} />
                );
              }
              if (route.name === 'More') {
                return (
                  <MaterialIcons
                    name="more-horiz"
                    size={25 * heightRatio}
                    color={color}
                  />
                );
              }
              if (route.name === 'Users') {
                iconName = 'ios-person-circle-sharp';
              } else if (route.name === 'Groups') {
                iconName = 'people';
              }

              // You can return any component that you like here!
              return (
                <Ionicons
                  name={iconName}
                  size={24 * heightRatio}
                  color={color}
                />
              );
            },
          })}
          tabBarOptions={{
            activeTintColor: theme.color.blue,
            inactiveTintColor: 'rgba(0,0,0,0.5)',
            activeBackgroundColor: theme.color.white,
            inactiveBackgroundColor: theme.color.white,
            labelStyle: { fontSize: 12 },
          }}>
          {tabs.isChatEnabled && (
            <Tab.Screen
              name="Chats"
              component={CometChatConversationListWithMessages}
            />
          )}
          {tabs.isUserListEnabled && (
            <Tab.Screen
              name="Users"
              component={CometChatUserListWithMessages}
            />
          )}
          {tabs.isGroupListEnabled && (
            <Tab.Screen
              name="Groups"
              component={CometChatGroupListWithMessages}
            />
          )}
          {tabs.isUserSettingsEnabled && (
            <Tab.Screen name="More" component={CometChatUserProfile} />
          )}
        </Tab.Navigator>
      ) : null}
    </CometChatContextProvider>
  );
}

export default CometChatUI;
