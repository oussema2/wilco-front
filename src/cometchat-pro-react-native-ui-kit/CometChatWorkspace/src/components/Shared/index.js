export { default as CometChatAvatar } from './CometChatAvatar';
export { default as CometChatSharedMedia } from './CometChatSharedMedia';
export { default as CometChatBadgeCount } from './CometChatBadgeCount';
export { default as CometChatBackdrop } from './CometChatBackdrop';
export { default as CometChatUserPresence } from './CometChatUserPresence';
