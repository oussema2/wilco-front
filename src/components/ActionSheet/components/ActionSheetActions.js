import React from 'react';
import { observer } from 'mobx-react';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { ActionSheetGradient } from '../../ActionSheetGradient';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetActions = () => {
	const actionSheetService = useActionSheetService();

	return (
		<ActionSheetGradient style={styles.actions}>
			{actionSheetService.actions.map( ( action ) => (
				<ActionSheetButton
					key={action.title}
					{...action}
					onPress={() => {
						actionSheetService.close();
						action.onPress();
					}}
				/>
			) )}
		</ActionSheetGradient>
	);
};

export default observer( ActionSheetActions );
