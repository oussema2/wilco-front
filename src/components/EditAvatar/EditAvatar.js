import React from 'react';
import { Pressable, View } from 'react-native';
import PropTypes from 'prop-types';

import ImagePicker from 'react-native-image-crop-picker';
import { styles } from './EditAvatar.styles';
import { Avatar } from '../Avatar';
import { TextButton } from '../TextButton';
import useGalleryPermissionHandler from '../../hooks/useGalleryPermissionHandler';

const EditAvatar = ( {
	testID, onAvatarChange, source, variant, size
} ) => {
	const _onImageSelected = ( image ) => {
		onAvatarChange( { uri: image.path, base64: image.data } );
	};

	const _openImagePicker = async () => {
		try {
			const image = await ImagePicker.openPicker( {
				mediaType: 'photo',
				cropping: true,
				cropperCircleOverlay: true,
				includeBase64: true,
				cropperChooseText: 'Save'
			} );
			_onImageSelected( image );
		} catch { }
	};

	const handleGalleryPermission = useGalleryPermissionHandler( _openImagePicker );
	const variantText = variant === 'user' ? 'profile' : variant;
	const buttonTitle = source ? `Change ${variantText} photo` : `Add ${variantText} photo`;

	return (
		<View style={styles.editAvatar} testID={testID}>
			<Pressable
				testID="edit-avatar-image-testID"
				onPress={() => handleGalleryPermission()}
			>
				<Avatar source={source} variant={variant} size={size} />
			</Pressable>
			<TextButton
				testID="edit-avatar-button-testID"
				onPress={() => handleGalleryPermission()}
				title={buttonTitle}
			/>
		</View>
	);
};

EditAvatar.propTypes = {
	testID: PropTypes.string,
	onAvatarChange: PropTypes.func.isRequired,
	source: PropTypes.shape( {
		uri: PropTypes.string.isRequired
	} ),
	variant: PropTypes.oneOf( [ 'user', 'aircraft' ] ),
	size: PropTypes.oneOf( [ 'big', 'medium', 'small' ] )
};

EditAvatar.defaultProps = {
	testID: 'EditAvatar-Component',
	source: null,
	variant: 'user',
	size: 'medium'
};

export default EditAvatar;
