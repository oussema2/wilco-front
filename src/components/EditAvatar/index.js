export { default as EditAvatar } from './EditAvatar';
