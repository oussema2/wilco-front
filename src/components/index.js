export { PrimaryButton } from './PrimaryButton';
export { TextInput } from './TextInput';
