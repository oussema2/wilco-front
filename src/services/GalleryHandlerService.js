import PermissionsHandlerService, {
	galleryPermission,
	permissionResults
} from './PermissionsHandlerService';
import AlertMessagesService from './AlertMessagesService';

let sharedInstance;

export default class GalleryHandlerService {
	constructor( {
		permissionHandlerService = PermissionsHandlerService.shared(),
		alertMessagesService = AlertMessagesService.shared()
	} = {} ) {
		this.permissionHandlerService = permissionHandlerService;
		this.alertMessagesService = alertMessagesService;
	}

	static shared = () => {
		if ( sharedInstance ) { return sharedInstance; }

		sharedInstance = new GalleryHandlerService();
		return sharedInstance;
	}

	async handleGalleryPermission( _callback ) {
		const requestResult = await this.permissionHandlerService.request( galleryPermission );

		const GRANTED_PERMISSION_RESULTS = [ permissionResults.granted, permissionResults.limited ];
		const permissionWasGranted = ( result ) => GRANTED_PERMISSION_RESULTS.includes( result );

		if ( permissionWasGranted( requestResult ) ) {
			_callback();
		} else {
			this._showAlertForOpenSettings();
		}
	}

	_showAlertForOpenSettings() {
		this.alertMessagesService.showConfirmationAlert( {
			title: '"Wilco" Would Like to Access Your Photos',
			message: 'This app requires access to the photo library.',
			cancelButton: { onPress: () => {} },
			confirmButton: {
				text: 'Open Settings',
				onPress: () => this.permissionHandlerService.openSettings()
			}
		} );
	}
}
