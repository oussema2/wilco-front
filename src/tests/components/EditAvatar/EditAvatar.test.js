import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { EditAvatar } from '../../../components/EditAvatar';
import GalleryHandlerService from '../../../services/GalleryHandlerService';

describe( '<EditAvatar />', () => {
	let component;
	const testID = 'edit-avatar-component-testID';
	const buttonTestID = 'edit-avatar-button-testID';
	const imageTestID = 'edit-avatar-image-testID';
	const onAvatarChange = jest.fn();

	const setUp = ( props ) => {
		component = render( <EditAvatar
			testID={testID}
			onAvatarChange={onAvatarChange}
			{...props}
		/> );
	};

	beforeEach( () => {
		setUp();
	} );

	describe( 'with the default props', () => {
		it( 'renders the EditAvatar component correctly', () => {
			expect( component.queryByTestId( testID ) ).toBeDefined();

			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'when the variant is "aircraft', () => {
		describe( 'when no source is provided', () => {
			it( 'renders the button with text "Add aircraft photo"', () => {
				setUp( { variant: 'aircraft' } );
				expect( component.queryByText( 'Add aircraft photo' ) ).toBeDefined();
				expect( component ).toMatchSnapshot();
			} );
		} );

		describe( 'when a source is provided', () => {
			it( 'renders the provided image source and the button with text "Change aircraft photo"', () => {
				setUp( { variant: 'aircraft', source: { uri: 'sample_uri' } } );
				expect( component.queryByText( 'Change aircraft photo' ) ).toBeDefined();
				expect( component ).toMatchSnapshot();
			} );
		} );
	} );

	describe( 'when the variant is "user"', () => {
		describe( 'when no source is provided', () => {
			it( 'renders the button with text "Add profile photo"', () => {
				setUp( { variant: 'user' } );
				expect( component.queryByText( 'Add profile photo' ) ).toBeDefined();
				expect( component ).toMatchSnapshot();
			} );
		} );

		describe( 'when a source is provided', () => {
			it( 'renders the provided image source and the button with text "Change profile photo"', () => {
				setUp( { variant: 'user', source: { uri: 'sample_uri' } } );
				expect( component.queryByText( 'Change profile photo' ) ).toBeDefined();
				expect( component ).toMatchSnapshot();
			} );
		} );
	} );

	describe( 'when user press the button', () => {
		it( 'calls to handleGalleryPermission', async () => {
			const galleryHandlerService = GalleryHandlerService.shared();
			const handleGalleryPermissionSpy = jest.spyOn( galleryHandlerService, 'handleGalleryPermission' );

			fireEvent.press( component.queryByTestId( buttonTestID ) );
			expect( handleGalleryPermissionSpy ).toHaveBeenCalled();
		} );
	} );

	describe( 'when user press the avatar', () => {
		it( 'calls to handleGalleryPermission', async () => {
			const galleryHandlerService = GalleryHandlerService.shared();
			const handleGalleryPermissionSpy = jest.spyOn( galleryHandlerService, 'handleGalleryPermission' );

			fireEvent.press( component.queryByTestId( imageTestID ) );
			expect( handleGalleryPermissionSpy ).toHaveBeenCalled();
		} );
	} );
} );
