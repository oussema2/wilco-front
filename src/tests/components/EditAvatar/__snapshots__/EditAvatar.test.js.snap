// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditAvatar /> when the variant is "aircraft when a source is provided renders the provided image source and the button with text "Change aircraft photo" 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "height": 165,
      "justifyContent": "space-between",
      "marginTop": 39,
    }
  }
  testID="edit-avatar-component-testID"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="edit-avatar-image-testID"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "aspectRatio": 1,
          "backgroundColor": "#A0A0A0",
          "borderRadius": 68,
          "display": "flex",
          "height": 44,
          "justifyContent": "center",
        }
      }
      testID="Avatar-Component"
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 68,
              "height": "100%",
              "width": "100%",
            },
          ]
        }
      >
        <FastImageView
          onFastImageLoadEnd={[Function]}
          resizeMode="cover"
          source={
            Object {
              "uri": "sample_uri",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          testID="image"
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "aspectRatio": 1,
              "backgroundColor": "#A0A0A0",
              "borderRadius": 68,
              "display": "flex",
              "height": 44,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "height": 18,
                  "marginLeft": 2,
                  "width": 24,
                },
              ]
            }
          >
            <FastImageView
              resizeMode="cover"
              source={
                Object {
                  "testUri": "../../../src/assets/icons/aircraft.png",
                }
              }
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
              testID="Avatar-Component-loader"
              tintColor="#FFFFFF"
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        undefined,
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="edit-avatar-button-testID"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-end",
            "backgroundColor": "transparent",
            "borderColor": "#2089dc",
            "borderRadius": 26,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 0,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#12A3BE",
              "fontFamily": "NotoSansJP-Bold",
              "fontSize": 15,
              "fontWeight": undefined,
              "letterSpacing": 0,
              "lineHeight": 20,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Change aircraft photo
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<EditAvatar /> when the variant is "aircraft when no source is provided renders the button with text "Add aircraft photo" 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "height": 165,
      "justifyContent": "space-between",
      "marginTop": 39,
    }
  }
  testID="edit-avatar-component-testID"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="edit-avatar-image-testID"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "aspectRatio": 1,
          "backgroundColor": "#A0A0A0",
          "borderRadius": 68,
          "display": "flex",
          "height": 44,
          "justifyContent": "center",
        }
      }
      testID="Avatar-Component"
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "height": 18,
              "marginLeft": 2,
              "width": 24,
            },
          ]
        }
      >
        <FastImageView
          onFastImageLoadEnd={[Function]}
          resizeMode="cover"
          source={
            Object {
              "testUri": "../../../src/assets/icons/aircraft.png",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          testID="image"
          tintColor="#FFFFFF"
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "aspectRatio": 1,
              "backgroundColor": "#A0A0A0",
              "borderRadius": 68,
              "display": "flex",
              "height": 44,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "height": 18,
                  "marginLeft": 2,
                  "width": 24,
                },
              ]
            }
          >
            <FastImageView
              resizeMode="cover"
              source={
                Object {
                  "testUri": "../../../src/assets/icons/aircraft.png",
                }
              }
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
              testID="Avatar-Component-loader"
              tintColor="#FFFFFF"
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        undefined,
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="edit-avatar-button-testID"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-end",
            "backgroundColor": "transparent",
            "borderColor": "#2089dc",
            "borderRadius": 26,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 0,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#12A3BE",
              "fontFamily": "NotoSansJP-Bold",
              "fontSize": 15,
              "fontWeight": undefined,
              "letterSpacing": 0,
              "lineHeight": 20,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Add aircraft photo
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<EditAvatar /> when the variant is "user" when a source is provided renders the provided image source and the button with text "Change profile photo" 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "height": 165,
      "justifyContent": "space-between",
      "marginTop": 39,
    }
  }
  testID="edit-avatar-component-testID"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="edit-avatar-image-testID"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "aspectRatio": 1,
          "backgroundColor": "#A0A0A0",
          "borderRadius": 68,
          "display": "flex",
          "height": 44,
          "justifyContent": "center",
        }
      }
      testID="Avatar-Component"
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 68,
              "height": "100%",
              "width": "100%",
            },
          ]
        }
      >
        <FastImageView
          onFastImageLoadEnd={[Function]}
          resizeMode="cover"
          source={
            Object {
              "uri": "sample_uri",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          testID="image"
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "aspectRatio": 1,
              "backgroundColor": "#A0A0A0",
              "borderRadius": 68,
              "display": "flex",
              "height": 44,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "aspectRatio": 1,
                  "height": 24,
                  "width": 32,
                },
              ]
            }
          >
            <FastImageView
              resizeMode="cover"
              source={
                Object {
                  "testUri": "../../../src/assets/icons/user.png",
                }
              }
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
              testID="Avatar-Component-loader"
              tintColor="#FFFFFF"
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        undefined,
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="edit-avatar-button-testID"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-end",
            "backgroundColor": "transparent",
            "borderColor": "#2089dc",
            "borderRadius": 26,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 0,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#12A3BE",
              "fontFamily": "NotoSansJP-Bold",
              "fontSize": 15,
              "fontWeight": undefined,
              "letterSpacing": 0,
              "lineHeight": 20,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Change profile photo
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<EditAvatar /> when the variant is "user" when no source is provided renders the button with text "Add profile photo" 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "height": 165,
      "justifyContent": "space-between",
      "marginTop": 39,
    }
  }
  testID="edit-avatar-component-testID"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="edit-avatar-image-testID"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "aspectRatio": 1,
          "backgroundColor": "#A0A0A0",
          "borderRadius": 68,
          "display": "flex",
          "height": 44,
          "justifyContent": "center",
        }
      }
      testID="Avatar-Component"
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "aspectRatio": 1,
              "height": 24,
              "width": 32,
            },
          ]
        }
      >
        <FastImageView
          onFastImageLoadEnd={[Function]}
          resizeMode="cover"
          source={
            Object {
              "testUri": "../../../src/assets/icons/user.png",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          testID="image"
          tintColor="#FFFFFF"
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "aspectRatio": 1,
              "backgroundColor": "#A0A0A0",
              "borderRadius": 68,
              "display": "flex",
              "height": 44,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "aspectRatio": 1,
                  "height": 24,
                  "width": 32,
                },
              ]
            }
          >
            <FastImageView
              resizeMode="cover"
              source={
                Object {
                  "testUri": "../../../src/assets/icons/user.png",
                }
              }
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
              testID="Avatar-Component-loader"
              tintColor="#FFFFFF"
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        undefined,
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="edit-avatar-button-testID"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-end",
            "backgroundColor": "transparent",
            "borderColor": "#2089dc",
            "borderRadius": 26,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 0,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#12A3BE",
              "fontFamily": "NotoSansJP-Bold",
              "fontSize": 15,
              "fontWeight": undefined,
              "letterSpacing": 0,
              "lineHeight": 20,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Add profile photo
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<EditAvatar /> with the default props renders the EditAvatar component correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "height": 165,
      "justifyContent": "space-between",
      "marginTop": 39,
    }
  }
  testID="edit-avatar-component-testID"
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="edit-avatar-image-testID"
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "aspectRatio": 1,
          "backgroundColor": "#A0A0A0",
          "borderRadius": 68,
          "display": "flex",
          "height": 44,
          "justifyContent": "center",
        }
      }
      testID="Avatar-Component"
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "aspectRatio": 1,
              "height": 24,
              "width": 32,
            },
          ]
        }
      >
        <FastImageView
          onFastImageLoadEnd={[Function]}
          resizeMode="cover"
          source={
            Object {
              "testUri": "../../../src/assets/icons/user.png",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          testID="image"
          tintColor="#FFFFFF"
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "aspectRatio": 1,
              "backgroundColor": "#A0A0A0",
              "borderRadius": 68,
              "display": "flex",
              "height": 44,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "aspectRatio": 1,
                  "height": 24,
                  "width": 32,
                },
              ]
            }
          >
            <FastImageView
              resizeMode="cover"
              source={
                Object {
                  "testUri": "../../../src/assets/icons/user.png",
                }
              }
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
              testID="Avatar-Component-loader"
              tintColor="#FFFFFF"
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "borderRadius": 3,
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 3,
        },
        undefined,
        false,
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "busy": false,
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="edit-avatar-button-testID"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-end",
            "backgroundColor": "transparent",
            "borderColor": "#2089dc",
            "borderRadius": 26,
            "borderWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 0,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#12A3BE",
              "fontFamily": "NotoSansJP-Bold",
              "fontSize": 15,
              "fontWeight": undefined,
              "letterSpacing": 0,
              "lineHeight": 20,
              "paddingVertical": 1,
              "textAlign": "center",
            }
          }
        >
          Add profile photo
        </Text>
      </View>
    </View>
  </View>
</View>
`;
